src/stores/modules/User.store.js

```diff
@@ -38,13 +42,24 @@ const actions = {
   },
   async REGISTER ({ commit }, payload) {
 
-    const result = await api.user.register(payload.username, payload.password);
+    const result = await api.user.register(payload.username, payload.password, payload.email);
 
     if(result.success) {
-      api.user.persistUser(result.userData.username,result.token);
+      api.user.persistUser(result.userData.username, result.token, result.userData.email);
       commit('receiveLogin', result);
     }
-  }
+  },
+  async UPDATE ({ commit }, payload) {
+    const result = await api.user.updateFields(payload);
+    if (result.success) {
```

Comment:
FYI, Errors should automatically get picked up and displayed by the fetch method based on the status code. Leave as is for now as there's bug raised to cover this.